src/Node/Buffer/Internal.purs

```diff
@@ -30,81 +31,102 @@ import Prelude
 
 import Data.ArrayBuffer.Types (ArrayBuffer)
 import Data.Maybe (Maybe)
+import Effect.Uncurried (EffectFn1, EffectFn3, EffectFn4, EffectFn5, runEffectFn1, runEffectFn3, runEffectFn4, runEffectFn5)
+import Node.Buffer.Class (freeze, thaw)
 import Node.Buffer.Immutable (ImmutableBuffer)
 import Node.Buffer.Immutable as Immutable
 import Node.Buffer.Types (BufferValueType, Octet, Offset)
 import Node.Encoding (Encoding, encodingToNode)
 import Unsafe.Coerce (unsafeCoerce)
 
-unsafeFreeze :: forall buf m. Monad m => buf -> m ImmutableBuffer
+unsafeFreeze :: Buffer -> Effect ImmutableBuffer
 unsafeFreeze = pure <<< unsafeCoerce
 
-unsafeThaw :: forall buf m. Monad m => ImmutableBuffer -> m buf
+unsafeThaw :: ImmutableBuffer -> Effect Buffer
 unsafeThaw = pure <<< unsafeCoerce
 
-usingFromImmutable :: forall buf m a. Monad m => (ImmutableBuffer -> a) -> buf -> m a
+usingFromImmutable :: forall a. (ImmutableBuffer -> a) -> Buffer -> Effect a
 usingFromImmutable f buf = f <$> unsafeFreeze buf
 
-usingToImmutable :: forall buf m a. Monad m => (a -> ImmutableBuffer) -> a -> m buf
+usingToImmutable :: forall a. (a -> ImmutableBuffer) -> a -> Effect Buffer
 usingToImmutable f x = unsafeThaw $ f x
 
-create :: forall buf m. Monad m => Int -> m buf
+create :: Int -> Effect Buffer
 create = usingToImmutable Immutable.create
 
-foreign import copyAll :: forall a buf m. a -> m buf
```

Comment:
Note: this was used to implement both `freeze` and `thaw`.

Reply:
:+1: I generally think replacing functions like this with FFI for separate operations is better as it makes the interface more portable. Probably doesn't really apply to `-node-buffer`, but just thought I'd throw it out there! :smile: